synapse/lib/layer.py

```diff
@@ -4756,36 +4754,36 @@ async def confirmLayerEditPerms(self, user, gateiden, delete=False):
         # tags
         # NB: tag perms should be yielded for every leaf on every node in the layer
         if not allow_tags:
-            async with self.core.getSpooledDict() as tags:
-                async for lkey, buid in s_coro.pause(self.layrslab.scanByPref(INDX_TAG, db=self.indxdb)):
-                    abrv = lkey[2:]
-                    abrvs = list(tags.get(buid, []))
-                    abrvs.append(abrv)
-                    await tags.set(buid, abrvs)
+            async with self.core.getSpooledDict() as tagdict:
+                async for byts, abrv in s_coro.pause(self.core.indxabrv.iterByPref(INDX_TAG)):
+                    tag = s_msgpack.un(byts[2:])[1]
+                    async for _, nid in s_coro.pause(self.layrslab.scanByPref(abrv, db=self.indxdb)):
+                        tags = list(tagdict.get(nid, []))
+                        if tag in tags:
+                            continue
```

Comment:
The byts contain (form, tag) and you can skip abrvs where form=None, which will also remove the need for the `if tag in tags` check